File: projects/engineering.purdue.edu/kak/scriptingwithobjects/swocode/chap15/BroadcastClient.py
#!/usr/bin/env python

###  BroadcastClient.py

import socket                                                        #(A)

MAX_MSG_LEN = 1024                                                   #(B)

sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)              #(C)   
sock.bind( ('', 9000) )                                              #(D)

while 1:                                                             #(E)
    mesg_in = sock.recv( MAX_MSG_LEN )                               #(F)
    print mesg_in                                                    #(G)

File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap15/BroadcastClient.pl
#!/usr/bin/perl -w
use strict;

###  BroadcastClient.pl

use IO::Socket;                                                      #(A)
use constant MAX_MSG_LEN => 1024;                                    #(B)

my $sock = IO::Socket::INET->new( LocalPort  => '9000',              #(C)
                                  Proto     => "udp",                #(D)
                                )
                   or die $@;                                        #(E)
my $mesg_in;                                                         #(F)
while (1) {                                                          #(G)
    $sock->recv( $mesg_in, MAX_MSG_LEN ) or die "recv(): $!";        #(H)
    print "$mesg_in\n";                                              #(I)
}
